redfish-core/lib/pcie_slots.hpp

```diff
@@ -346,10 +346,37 @@ void checkPCIeSlotsCount(
         });
 }
 
+inline void getLocationCode(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                            const size_t index,
+                            const std::string& connectionName,
+                            const std::string& pcieSlotPath)
+{
+    sdbusplus::asio::getProperty<std::string>(
+        *crow::connections::systemBus, connectionName, pcieSlotPath,
+        "xyz.openbmc_project.Inventory.Decorator.LocationCode", "LocationCode",
+        [asyncResp, index](const boost::system::error_code& ec1,
+                           const std::string& property) {
+        if (ec1)
+        {
+            BMCWEB_LOG_ERROR << "Can't get location code property for PCIeSlot";
+            messages::internalError(asyncResp->res);
+            return;
+        }
+        if (property.empty())
+        {
+            BMCWEB_LOG_ERROR << "PcieSlot location code value is empty ";
```

Comment:
probably shouldn't be a LOG_ERROR if we aren't throwing an error. I guess oh well. WARNING makes more sense.